astropy/units/core.py

```diff
@@ -2353,7 +2352,7 @@ def __init__(
                     for p in unit.powers
                 ]
 
-            self._scale = sanitize_scale(scale)
+            self._scale = 1.0 if is_effectively_unity(scale) else scale
```

Comment:
This is not quite good enough, as it removes the checks on nearly perfectly real or imaginary complex input (which clearly is not tested well enough...)

Reply:
This is good enough to check nearly perfectly real complex numbers. It is true that this does not check nearly perfectly imaginary numbers, but is that big enough of a problem that it deserves a new function?

Reply:
Not really a new function, but one stripped to actually sanitizing the scale rather than also checking its validity...